### DISCUSSION TITLE
Modifying thermostat week programs from HA

### DISCUSSION THREAD

**Discussion by hosswald** [SukramJ/aiohomematic#1732] [Ideas]
I did not find anything yet related to this topic, but I assume it's a common question people who use Homematic thermostats might have...

I have a combination of two HmIP-FALMOT-C16 floor heating controllers and multiple HmIP-BWTH and HmIP-WTH wall thermostats directly connected to the FALMOT, as well as a few radiator thermostats.
Programming the thermostats in RaspberryMatic/ths CCU is not a great experience - the week programs have to be programmed for each thermostat individually, there is no graphical representation etc. It's not something I would want to touch again after setting it up once.
The advantage however is that all features of Homematic work, especially "Optimum start stop", which starts heating early to reach the programmed temperature at the specified time. Homematic also supports dynamic hydraulic balancing.

The alternative I tried is the Scheduler card (https://github.com/nielsfaber/scheduler-component). It's much nicer to set up and change, but I'm pretty sure optimum start stop won't work with it and I'm unsure about hydraulic balancing. Also, it will stop working if the CCU or Home Assistant become unavailable, while week schedules are copied to the thermostats as far as I understand.

Is there any possibility that the schedules (week programs) of the thermostats in the CCU could be exposed and modified to/from Home Assistant, e.g. by exposing them as Schedules (https://www.home-assistant.io/integrations/schedule/)?

Thanks for the great work to all devs!

**Comment by SukramJ** [comment]
I don't have any ambitions to add this to the integration, even if it were possible.

**Reply by SukramJ** [reply]
I changed my mind a little and at least included the necessary HA actions. See #1761
